### ISSUE TITLE
Is this real?

### ISSUE THREAD

**Issue by OliverEvans96** [pallada-92/dna-3d-engine#3]
Forgive my ignorance, I have almost no chemistry background, I just stumbled across this repo, and I'm pretty mind-blown.

You seem like a serious chemist, so I'll assume that it is real, but I had no idea that DNA could perform calculations. I don't know if you've had a chance to actually carry out the experiment yet, but I'd love to see a photo of the final result! Also, I'd love to hear what kind of practical applications (if any) you can imagine for this type of biological computation. There must be some fundamental differences from normal computer computations, no? It would be great if you could recommend a non-technical introduction to this stuff.

Thanks for sharing your crazy work!
Oliver

TLDR; pics or it didn't happen ;)

**Comment by pallada-92** [comment]
Hi, Olver!

It is more like a thought experiment now. It will require lots of additional efforts and money to make it happen in reality. 

Actually, I have maths/computer science background and have very superficial knowledge about biology, especially about wet lab experiments. But I've read [a paper](https://www.pnas.org/content/107/12/5393), that proposed a way to implement any chemical reaction network in DNA. So I did my best as a mathematician and software engineer to compose specific set of reactions with desired behavior and produced DNA with the compiler, developed as a part of that paper.

This particular type of computation can be done "in vitro" only, not in living cells. So I am not aware of any immediate practical applications to this science.

The computational architecture of chemical reaction networks is totally different from any other, that I know. You can learn about Petri nets as an introduction to this topic.

**Comment by OliverEvans96** [comment]
Thanks for the reply - it seems like an exciting thought experiment! I'm looking forward to seeing where it goes
